### ISSUE TITLE
Allow to use Supplier from java8 and guava as Randomizer

### ISSUE THREAD

**Issue by Toilal** [benas/random-beans#28]
Create a new PopulatorBuilder method supporting Supplier instead of Randomizer.

```java
class PopulatorBuilder8 extends PopulatorBuilder {
    public PopulatorBuilder registerRandomizer(final Class type, final Class fieldType, final String fieldName, final Supplier<?> randomizer) {
        return super.registerRandomizer(type, fieldType, fieldName, Java8Adapters.adapt(randomizer)):
    }
}
```

Same idea for guava Supplier.

It requires #26 to keep support of java version. (jpopulator-java8 and jpopulator-guava modules)

**Comment by benas** [comment]
Indeed, great idea :+1:  Thank you Rémi!

Let's plan that for next release :smiley:

**Comment by benas** [comment]
I was trying to add support for `Supplier` in the Builder but this would make the library require Java 8 as minimum Java version (while some people are still using Java 6, example in #43 ) .

I was not able to use java source level to 1.8 and target level to 1.6 (because only `random-beans-java8` module requires Java 8). So I need to pay attention to not use Java 8 APIs outside this module..

As of today, the library is compiled with Java 8 but the binary is still Java 6 compatible.

**Comment by PascalSchumacher** [comment]
I have started work an a pull request to add support for java 8 suppliers. It should be complete in a few days.

**Comment by PascalSchumacher** [comment]
pull request which adds the feature to use a java8 supplier: https://github.com/benas/random-beans/pull/96

**Comment by benas** [comment]
Hi @Toilal 

I've added support to use Java 8 suppliers as randomizers.

It would be great to give it a try if possible to see if it corresponds to what you have requested.
Many thanks upfront.

Kind regards
Mahmoud

**Comment by Toilal** [comment]
I think it's good thanks ! I'll effectively try later.

**Comment by benas** [comment]
Hi Rémi,

No problem, take your time.

This will be released soon in v3.0.0
In case of any issue please file a bug.

Note that support for the standard Java 8 API `Supplier` has been added. Support for guava API will not be included since it is specific.

Kind regards
Mahmoud
